crates/ere-hosts/src/main.rs

```diff
@@ -192,69 +183,32 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
-            for zkvm in zkvms {
-                run_benchmark(zkvm, &config, inputs.clone())?;
+
+            for zkvm in cli.zkvms {
+                let zkvm = get_zkvm_instance(
+                    &zkvm,
+                    &workspace_dir,
+                    Path::new("block-encoding-length"),
+                    resource.clone(),
+                )?;
+                run_benchmark(&zkvm, &config, inputs.clone())?;
             }
         }
     }
 
     Ok(())
 }
 
-fn get_zkvm_instances(
+fn get_zkvm_instance(
+    zkvm: &ErezkVM,
     workspace_dir: &Path,
     guest_relative: &Path,
     resource: ProverResourceType,
-) -> Result<Vec<Box<dyn zkVM + Sync>>, Box<dyn std::error::Error>> {
-    let mut name_zkvms: Vec<Box<dyn zkVM + Sync>> = Default::default();
-    #[allow(clippy::redundant_clone)]
-    {
-        #[cfg(feature = "sp1")]
-        {
-            run_cargo_patch_command("sp1", workspace_dir)?;
-            let program =
-                RV32_IM_SUCCINCT_ZKVM_ELF::compile(workspace_dir, &guest_relative.join("sp1"))?;
-            let zkvm = EreSP1::new(program, resource.clone());
-            name_zkvms.push(Box::new(zkvm));
-        }
-
-        #[cfg(feature = "zisk")]
-        {
-            run_cargo_patch_command("zisk", workspace_dir)?;
-            let program =
-                RV64_IMA_ZISK_ZKVM_ELF::compile(workspace_dir, &guest_relative.join("zisk"))?;
-            let zkvm = EreZisk::new(program, resource.clone());
-            name_zkvms.push(Box::new(zkvm));
-        }
-
-        #[cfg(feature = "risc0")]
-        {
-            run_cargo_patch_command("risc0", workspace_dir)?;
-            let program =
-                RV32_IM_RISC0_ZKVM_ELF::compile(workspace_dir, &guest_relative.join("risc0"))?;
-            let zkvm = EreRisc0::new(program, resource.clone());
-            name_zkvms.push(Box::new(zkvm));
-        }
-
-        #[cfg(feature = "openvm")]
-        {
-            run_cargo_patch_command("openvm", workspace_dir)?;
-            let program = OPENVM_TARGET::compile(workspace_dir, &guest_relative.join("openvm"))?;
-            let zkvm = EreOpenVM::new(program, resource.clone())?;
-            name_zkvms.push(Box::new(zkvm));
-        }
-
-        #[cfg(feature = "pico")]
-        {
-            run_cargo_patch_command("pico", workspace_dir)?;
-            let program = PICO_TARGET::compile(workspace_dir, &guest_relative.join("pico"))?;
-            let zkvm = ErePico::new(program, resource.clone());
-            name_zkvms.push(Box::new(zkvm));
-        }
-    }
-    Ok(name_zkvms)
+) -> Result<EreDockerizedzkVM, Box<dyn std::error::Error>> {
+    run_cargo_patch_command(zkvm.as_str(), workspace_dir)?;
+    let program = EreDockerizedCompiler::new(*zkvm, workspace_dir)
+        .compile(&workspace_dir.join(guest_relative).join(zkvm.as_str()))?;
+    Ok(EreDockerizedzkVM::new(*zkvm, program, resource.clone())?)
```

Comment:
@han0110, I realized that before we had some vector construction that I'm not sure was correct, since ony the last `run_cargo_patch_command` will survive in the `ere-guests` workspace patching, no? (asking to double check if makes sense to you).

I moved this to be single-zkVM construction, and call it inside the loop above in the file.